java/vector/src/main/codegen/templates/ArrowType.java

```diff
@@ -246,6 +251,61 @@ public boolean equals(Object obj) {
   }
   </#list>
 
+  /**
+   * A user-defined data type that wraps an underlying storage type.
+   */
+  public abstract static class ExtensionType extends ComplexType {
+    /** The on-wire type for this user-defined type. */
+    public abstract ArrowType storageType();
+    /** The name of this user-defined type. Used to identify the type during serialization. */
+    public abstract String extensionName();
+    /** Check equality of this type to another user-defined type. */
+    public abstract boolean extensionEquals(ExtensionType other);
+    /** Save any metadata for this type. */
+    public abstract String serialize();
```

Comment:
it seems like it should be bytes, but I'm not sure what is exposed in the flatbuffer interface.

Reply:
The metadata is a string-string map, so String is expected here (even though it does feel weird).